CMakeLists.txt

```diff
@@ -111,10 +111,10 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_STANDARD 17)
 if(MSVC)
     string(APPEND CMAKE_CXX_FLAGS " /EHsc")
-    if(CMAKE_BUILD_TYPE STREQUAL "Release")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zi /guard:cf")
-        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /DEBUG /debugtype:cv,fixup /guard:cf")
-    endif()
+    string(APPEND CMAKE_CXX_FLAGS_RELEASE " /Zi /guard:cf /Os")
+    string(APPEND CMAKE_EXE_LINKER_FLAGS_RELEASE " /DEBUG /debugtype:cv,fixup /guard:cf")
+else()
+    string(REGEX REPLACE "-O[0-3]" "-Os" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
```

Comment:
If you want `MinSizeRel`, why do you build `Release`?

Reply:
fair point, I forgot about MinSizeRel

Reply:
We should probably suck out all these flag configurations into an "Official Build Toolchain"